develop/migration_mt_django.rst

```diff
@@ -88,84 +197,322 @@ Django automatically creates a simple "id" primary key on models, so we will nee
+  btree (project_id, manager_id)
+
+
+Drop this constraint in a migration:
+
+.. code-block:: python
+
+  from django.db import migrations
+
+  class Migration(migrations.Migration):
+
+    dependencies = [
+      # leave this as it was generated
+    ]
+
+    operations = [
+      migrations.RunSQL("""
+        DROP CONSTRAINT myapp_projectman_project_id_manager_id_bc477b48_uniq;
+      """),
+
+
+Then change your models to have a ``unique_together`` including the ``account\_id``
+
+
+.. code-block:: python
+
+  class ProjectManager(models.Model):
+      project = models.ForeignKey(Project, on_delete=models.CASCADE)
+      manager = models.ForeignKey(Manager, on_delete=models.CASCADE)
+      account = models.ForeignKey(Account, on_delete=models.CASCADE)
+
+      class Meta:
+          unique_together=(('account', 'project', 'manager'))
+
+
+And finally apply the changes by creating a new migration to generate these constraints:
+
+
+.. code-block:: python
+
+  python manage.py makemigrations
+
+
+3. Updating the models to use TenantModelMixin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, we'll use the `django-multitenant <https://github.com/citusdata/django-multitenant>`_ library to add account_id to foreign keys, and make application queries easier later on.
 
 In requirements.txt for your Django application, add
 
 ::
 
-  django_multitenant>=1.1.0
+  django_multitenant>=2.0.0
 
 Run ``pip install -r requirements.txt``.
 
+
 In settings.py, change the database engine to the customized engine provied by django-multitenant:
 
 .. code-block:: python
 
   'ENGINE': 'django_multitenant.backends.postgresql'
 
-Add a few more imports to your models file:
+
+
+**3.1 Introducing the TenantModelMixin and TenantManager**
+
+
+The models will now not only inherit from ``models.Model`` but also from the ``TenantModelMixin``.
+
+To do that in your :code:`models.py` file you will need to do the following imports
 
 .. code-block:: python
 
-  from django_multitenant.models import *
-  from django_multitenant.fields import *
+  from django_multitenant.mixins import *
+
 
-Change all the models to inherit from ``TenantModel`` rather than ``Model``, set the tenant\_id on each model, and use ``TenantForeignKey`` rather than ``ForeignKey`` for any foreign key which does not already contain the tenant\_id.
+Then your models should be changed to have the new manager and inheritance. As we are using Mixins, your models and managers can also inherits from other mixins you might have been using previously. It is for example compatible with using :code:`django.contrib.gis.db` models.
+
+You will also, at this point introduce the tenant_id, to define which column is the distribution column.
 
 .. code-block:: python
 
-  class Store(TenantModel):
-    name = models.CharField(max_length=255)
-    url = models.URLField()
-    tenant_id = "id"
+  class TenantManager(TenantManagerMixin, models.Manager):
```

Comment:
What does TenantManager do? Is this what you meant above when you said, "your models should be changed to have the new manager" ?

Reply:
The `TenantManager` uses the `TenantManagerMixin` it's basically what handles the changes in the ORM calls (like adding filters to joins and all). It's necessary for the models to do the right queries, and also it means that if they have their own manager, let's say they have 

```
class ProjectManager(models.Manager):
    def get_queryset(self):
        # do something
```

They can change it to be   


```
class ProjectManager(TenantManagerMixin, models.Manager):
    def get_queryset(self):
        # do something
```